prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/PredictionServiceImpl.java

```diff
@@ -160,4 +165,48 @@ public List<String> getPredictedProperties(NodeRef nodeRef)
                 .map(qName -> qName.toPrefixString(namespaceService))
                 .collect(toList());
     }
+
+    @Override
+    public void reviewPrediction(NodeRef predictionNodeRef, ReviewStatus reviewStatus) throws PredictionStateChangedException
+    {
+        ChildAssociationRef parentAssocRef = nodeService.getPrimaryParent(predictionNodeRef);
+        NodeRef parentNode = parentAssocRef.getParentRef();
+        Map<QName, Serializable> existingProperties = new HashMap<>(nodeService.getProperties(parentNode));
+        Prediction prediction = childAssocToPrediction(parentAssocRef);
+        QName propertyQName = QName.createQName(prediction.getProperty(), namespaceService);
+        if (existingProperties.get(propertyQName).equals(prediction.getPredictionValue()))
+        {
+            updatePredictionState(predictionNodeRef, parentNode, existingProperties, prediction, propertyQName, reviewStatus);
+        }
+        else
+        {
+            throw new PredictionStateChangedException(prediction.getProperty() + " property value has changed, prediction is no longer valid!");
+        }
```

Comment:
idk if everyone will agree with me, but I prefer not to have nested if-else structures, so I would suggest doing something like this:
```
if (!existingProperties.get(propertyQName).equals(prediction.getPredictionValue()))
{
throw new PredictionStateChangedException(prediction.getProperty() + " property value has changed, prediction is no longer valid!");
}

updatePredictionState(predictionNodeRef, parentNode, existingProperties, prediction, propertyQName, reviewStatus);
```

If you do this, you can also get rid of the `updatePredictionState` method and try to simplify it in a similar way

Reply:
I agree with Pawel - you should try to get rid of those nested conditionals.

Reply:
Good idea, refactored